src/mongo/db/query/explain_options_test.cpp

```diff
@@ -60,6 +60,18 @@ TEST(ExplainOptionsTest, ExplainOptionsSerializeToBSONCorrectly) {
                       ExplainOptions::toBSON(ExplainOptions::Verbosity::kExecAllPlans));
 }
 
+TEST(ExplainOptionsTest, InvalidOptionsIfAgumentNotRecognized) {
+    ASSERT_EQ(ExplainOptions::parseCmdBSON(fromjson("{explain: {}, foo: 1}")).getStatus(),
+              ErrorCodes::InvalidOptions);
+    ASSERT_EQ(
+        ExplainOptions::parseCmdBSON(fromjson("{explain: {}, verbosity: 'queryPlanner', foo: 1}"))
+            .getStatus(),
+        ErrorCodes::InvalidOptions);
+    ASSERT_EQ(
+        ExplainOptions::parseCmdBSON(fromjson("{explain: {}, verosity: {foo: 'bar'}}")).getStatus(),
+        ErrorCodes::InvalidOptions);
+}
+
```

Comment:
For completeness can you add a test that should work and not assert?

Reply:
I have added a non-asserting test at https://github.com/mongodb/mongo/pull/1315/commits/ce74977927ddf3f24bfd2eb17ce6e97a6b9d121c#diff-a4bfb6c740b4df06e6e94c7d256a3bbbR65, not sure if it is meaningful.